### DISCUSSION TITLE
"%lld%%" and similar files

### DISCUSSION THREAD

**Discussion by 1kalina** [pixelspark/sushitrain#6] [Ideas]
Hello, I want to do a localization to my native language. But I can't figure out what `%lld%%` is for? 
I see that `%lld%%` is the same for the supported languages. But, by the way, what is that?

Thank you for your work so much!

**Comment by pixelspark** [comment]
Great to hear you want to try and translate! 

The `%lld%%` and similar are placeholders that represent spaces in the text where the app can insert numbers or other pieces of text that may change at runtime. As an example, the string `Background synchronization ran for %lld seconds` may be shown in the app as `Background synchronization ran for 42 seconds`.

Note that translation may be quite a big task. If you want I can [generate a basic translation using ChatGPT](https://github.com/pixelspark/sushitrain/blob/main/localize.mjs) for the language you want, and send you the file, so you can improve it.

**Reply by acolomb** [reply]
Would you be interested in putting the project up for translation on Hosted Weblate, as we have done for the Android app and the core Syncthing? I could assist with getting the integration ready. We're certainly not the first to try translating an iOS app, so I expect it to be well supported.

**Reply by pixelspark** [reply]
Sure, I don't see any reason not to. I am not too familiar with how this works, just let me know what would be needed from my side (do we get automated PRs here for updated translations?)

**Reply by 1kalina** [reply]
> Note that translation may be quite a big task. If you want I can [generate a basic translation using ChatGPT](https://github.com/pixelspark/sushitrain/blob/main/localize.mjs) for the language you want, and send you the file, so you can improve it. 


Oh, I make it step by step in VSCode... but if ChatGPT can generate one file with all supported languages, it would be nice. I will check translation.
There is German and English there.., so I want to translate into Ukrainian.

**Reply by pixelspark** [reply]
@1kalina I will try to generate a file that has AI-translations for Ukrainian, I'll let you know when ready, so you can have a look at it.

**Reply by pixelspark** [reply]
@1kalina Please have a look at https://github.com/pixelspark/sushitrain/blob/translation/ukr/Localizable.xcstrings

**Comment by pixelspark** [comment]
Also note that `%%` means that you want to have a literal `%` in the text. This is because the `%` sign is otherwise interpreted as a special character. So the text `%lld%% completed` may show as `42% completed`. Here `%lld` means 'some number the app provides' and `%%` means 'just a percentage sign'.
